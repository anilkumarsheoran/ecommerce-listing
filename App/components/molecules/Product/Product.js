import PropTypes from "prop-types";
import style from "./Product.style";
import withStyle from "../../../global/withStyle";
import Image from "../../atoms/Image";
import { exclusive, sale } from "../../../locales/en-US";
import { staticPath } from "../../../constants/index";

/**
 *  Product component can be used to display a Product. This component is connected with Styled component
 *  @param {function} item : Item object contains information individual product information
 */

const Product = props => {
  const { item, className } = props;
  return (
    <li className={className}>
      <Image src={`${staticPath}${item.productImage}`} alt={item.productName} />
      <div>
        {item.isExclusive && <div className="exclusive-tag">{exclusive} </div>}
        {item.isSale && <div className="sale-tag">{sale} </div>}
        <div className="product-description">
          <div className="product-name">{item.productName}</div>
          <div className="product-price">{item.price}</div>
        </div>
      </div>
    </li>
  );
};

Product.propTypes = {
  item: PropTypes.object,
  className: PropTypes.string
};

export default withStyle(Product, style);
