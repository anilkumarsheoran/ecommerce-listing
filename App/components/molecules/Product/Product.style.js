import { css } from "styled-components";
import theme from "../../../styles/theme";
import fonts from "../../../styles/typography";

export default css`
  border: 1px solid ${theme.colors.lightGray};
  list-style: none;
  display: flex;
  flex: 1 1 300px;
  flex-wrap: wrap;
  padding: 30px 10px;
  padding-top: 20px;
  flex-direction: column;
  justify-content: space-between;
  flex-grow: 0;

  .sale-tag {
    background-color: ${theme.colors.red};
    color: ${theme.colors.white};
    padding: 16px 21px;
    display: inline-block;
  }

  .exclusive-tag {
    background-color: ${theme.colors.green};
    color: ${theme.colors.white};
    padding: 16px 35px;
    display: inline-block;
  }

  .product-description {
    display: flex;
    justify-content: space-between;
    margin-top: 30px;
    align-items: baseline;
  }
  .product-name {
    font-size: ${fonts.productName};
    max-width: 170px;
    font-weight: bold;
  }
  .product-price {
    font-size: ${fonts.productPrice};
    font-weight: bold;
  }
`;
