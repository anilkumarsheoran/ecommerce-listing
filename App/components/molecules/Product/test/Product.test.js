import { shallow } from "enzyme";
import { Product } from "../Product";
const { configure } = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

configure({ adapter: new Adapter() });

const props = {
  item: {
    index: 0,
    isSale: true,
    isExclusive: false,
    price: "$18.88",
    productImage: "product-1.jpg",
    productName: "Striped shirt",
    size: ["S", "L", "XL"]
  }
};

describe("Product Component", () => {
  test("should render correctly", () => {
    const product = shallow(<Product {...props} />);
    expect(product).toMatchSnapshot();
  });
});
