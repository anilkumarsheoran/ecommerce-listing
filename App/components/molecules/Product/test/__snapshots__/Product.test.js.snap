// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    item={
      Object {
        "index": 0,
        "isExclusive": false,
        "isSale": true,
        "price": "$18.88",
        "productImage": "product-1.jpg",
        "productName": "Striped shirt",
        "size": Array [
          "S",
          "L",
          "XL",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(Image)
          alt="Striped shirt"
          src="/static/product-1.jpg"
        />,
        <div>
          <div
            className="sale-tag"
          >
            Sale
             
          </div>
          <div
            className="product-description"
          >
            <div
              className="product-name"
            >
              Striped shirt
            </div>
            <div
              className="product-price"
            >
              $18.88
            </div>
          </div>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Striped shirt",
          "src": "/static/product-1.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="sale-tag"
            >
              Sale
               
            </div>,
            <div
              className="product-description"
            >
              <div
                className="product-name"
              >
                Striped shirt
              </div>
              <div
                className="product-price"
              >
                $18.88
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sale",
                " ",
              ],
              "className": "sale-tag",
            },
            "ref": null,
            "rendered": Array [
              "Sale",
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="product-name"
                >
                  Striped shirt
                </div>,
                <div
                  className="product-price"
                >
                  $18.88
                </div>,
              ],
              "className": "product-description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Striped shirt",
                  "className": "product-name",
                },
                "ref": null,
                "rendered": "Striped shirt",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$18.88",
                  "className": "product-price",
                },
                "ref": null,
                "rendered": "$18.88",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(Image)
            alt="Striped shirt"
            src="/static/product-1.jpg"
          />,
          <div>
            <div
              className="sale-tag"
            >
              Sale
               
            </div>
            <div
              className="product-description"
            >
              <div
                className="product-name"
              >
                Striped shirt
              </div>
              <div
                className="product-price"
              >
                $18.88
              </div>
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Striped shirt",
            "src": "/static/product-1.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <div
                className="sale-tag"
              >
                Sale
                 
              </div>,
              <div
                className="product-description"
              >
                <div
                  className="product-name"
                >
                  Striped shirt
                </div>
                <div
                  className="product-price"
                >
                  $18.88
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sale",
                  " ",
                ],
                "className": "sale-tag",
              },
              "ref": null,
              "rendered": Array [
                "Sale",
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="product-name"
                  >
                    Striped shirt
                  </div>,
                  <div
                    className="product-price"
                  >
                    $18.88
                  </div>,
                ],
                "className": "product-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Striped shirt",
                    "className": "product-name",
                  },
                  "ref": null,
                  "rendered": "Striped shirt",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$18.88",
                    "className": "product-price",
                  },
                  "ref": null,
                  "rendered": "$18.88",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
